src/backend/distributed/operations/shard_rebalancer.c

```diff
@@ -451,9 +526,22 @@ citus_shard_cost_by_disk_size(PG_FUNCTION_ARGS)
 	bool raiseErrors = true;
 	char *sizeQuery = PG_TOTAL_RELATION_SIZE_FUNCTION;
 	ShardInterval *shardInterval = LoadShardInterval(shardId);
-	List *colocatedShardList = ColocatedShardIntervalList(shardInterval);
-	StringInfo tableSizeQuery = GenerateSizeQueryOnMultiplePlacements(colocatedShardList,
-																	  sizeQuery);
+	char *shardName = ConstructQualifiedShardName(shardInterval);
+	StringInfo tableSizeQuery;
+
+	if (PartitionedTable(shardInterval->relationId))
```

Comment:
I don't think this is enough/correct. Before this change, we were collecting all the colocated shardIntervals. Now, if it is a partitioned table, we get only the size of  `its` partitions. Plus, if the input shardId is a colocated table to a partitioned table, we'd create the giant string anyway in the else part.

So, we should probably write a function like `ColocatedShardIntervalList()`, which skips shards of partitions